lora-phy/src/sx1276_7_8_9/mod.rs

```diff
@@ -544,10 +550,40 @@ where
     }
 
     async fn get_rx_packet_status(&mut self) -> Result<PacketStatus, RadioError> {
-        let rssi_raw = self.read_register(Register::RegPktRssiValue).await?;
-        let rssi = (rssi_raw as i16) - 157i16; // or -164 for low frequency port ???
-        let snr_raw = self.read_register(Register::RegPktRssiValue).await?;
-        let snr = snr_raw as i16;
+        let snr = {
+            let packet_snr = self.read_register(Register::RegPktSnrValue).await?;
+            packet_snr as i8 as i16 / 4
+        };
+
+        let rssi = {
+            let packet_rssi = self.read_register(Register::RegPktRssiValue).await?;
+
+            let rssi_offset = match self.config.chip {
+                Sx127xVariant::Sx1272 => SX1272_RSSI_OFFSET,
+                Sx127xVariant::Sx1276 => {
+                    let frequency_in_hz = {
+                        let msb = self.read_register(Register::RegFrfMsb).await? as u32;
+                        let mid = self.read_register(Register::RegFrfMid).await? as u32;
+                        let lsb = self.read_register(Register::RegFrfLsb).await? as u32;
+                        let frf = (msb << 16) + (mid << 8) + lsb;
+                        (frf as f64 * FREQUENCY_SYNTHESIZER_STEP) as u32
+                    };
+
+                    if frequency_in_hz > SX1276_RF_MID_BAND_THRESH {
+                        SX1276_RSSI_OFFSET_HF
+                    } else {
+                        SX1276_RSSI_OFFSET_LF
+                    }
+                }
+            };
+
+            if snr >= 0 {
```

Comment:
Both sx1272 and sx1276 SDK seem to do following:
```
if snr >= 0 {
    rssi_offset + packet_rssi + (packet_rssi >> 4)
} else {
    rssi_offset + packet_rssi + (packet_rssi >> 4) + snr
}
```

Reply:
Yes, I knew they shift 4 bits while the datasheets say that when SNR >= 0 it's packet_rssi * 16/15 for both sx1272 and sx1276. With the bit shift, It's equals to packet_rssi * 17/16 and I think the difference is negligible, but they add (packet_rssi >> 4) even when SNR < 0 which might be wrong I suppose? Should we follow the SDKs anyway?

Reply:
I don't know, to be honest. Current implementation is good enough, and as "perfect is the enemy of the good", we can just start using this.

Thank you again :100: